### PR TITLE
api: Removal of v1 API

### PR DESCRIPTION
**Pull request by derekargueta**
Description: This PR removes code related to v1 of the Envoy/UDPA API. This includes
- tests using v1 configurations
- JSON schemas for v1
- translation code from v1 to v2

Risk Level: medium (very large)
Testing: Existing
Docs Changes: N/A - the docs seem to have already been deleted.
Release Notes: done
Fixes: part of #2832

### CODE REVIEW

**Comment by derekargueta** [comment]
As mentioned in issue #5054, looks like we'll have to a dance with envoy-filter-example. One way we could do this is
1. Leave the `createFilterFactory` methods in the interface, remove `PURE` and provide a default empty implementation (and keep everything else in this PR)
2. in envoy-filter-example, remove the `createFilterFactory` methods, which will fall back the the above default implementation
3. remove `createFilterFactory` from Envoy

File: test/common/tcp_proxy/BUILD [hunk 1]
```diff
@@ -8,31 +8,30 @@ load(
...
+#        "//test/mocks/buffer:buffer_mocks",
+#        "//test/mocks/network:network_mocks",
+#        "//test/mocks/runtime:runtime_mocks",
+#        "//test/mocks/server:server_mocks",
+#        "//test/mocks/ssl:ssl_mocks",
+#        "//test/mocks/stream_info:stream_info_mocks",
+#        "//test/mocks/upstream:host_mocks",
+#        "//test/mocks/upstream:upstream_mocks",
+#    ],
+#)
```
Comment (derekargueta) [inline test/common/tcp_proxy/BUILD]: This will be fixed once https://github.com/envoyproxy/envoy/pull/8486 is merged

**Comment by alyssawilk** [comment]
@mattklein123 we're waiting until after the release until we land this yeah?  No problem with pre-review but just checking.

**Comment by mattklein123** [comment]
> @mattklein123 we're waiting until after the release until we land this yeah? No problem with pre-review but just checking.

Yeah

**Comment by derekargueta** [comment]
coverage at 97.47% (up from 97.10%) 🎉

File: include/envoy/server/filter_config.h [hunk 2]
```diff
@@ -311,22 +311,21 @@ class NamedNetworkFilterConfigFactory : public ProtocolOptionsFactory {
...
    * @return Network::FilterFactoryCb the factory creation function.
    */
-  virtual Network::FilterFactoryCb createFilterFactory(const Json::Object& config,
-                                                       FactoryContext& context) PURE;
-
-  /**
-   * v2 variant of createFilterFactory(..), where filter configs are specified as proto. This may be
-   * optionally implemented today, but will in the future become compulsory once v1 is deprecated.
-   */
   virtual Network::FilterFactoryCb createFilterFactoryFromProto(const Protobuf::Message& config,
```
[the remark this answers is not in the document]
Comment (derekargueta) [reply include/envoy/server/filter_config.h]: If the release is going out today/tomorrow, I'm fine letting this wait until after it's cut.

File: include/envoy/server/filter_config.h [hunk 3]
```diff
@@ -422,8 +418,7 @@ class NamedHttpFilterConfigFactory : public ProtocolOptionsFactory {
   /**
    * @return ProtobufTypes::MessagePtr create empty config proto message for v2. The filter
    *         config, which arrives in an opaque google.protobuf.Struct message, will be converted to
-   *         JSON and then parsed into this empty proto. Optional today, will be compulsory when v1
-   *         is deprecated.
+   *         JSON and then parsed into this empty proto.
    */
   virtual ProtobufTypes::MessagePtr createEmptyConfigProto() { return nullptr; }
```
[the remark this answers is not in the document]
Comment (derekargueta) [reply include/envoy/server/filter_config.h]: Yes I think it'd be a good idea to remove the default implementation and make these `PURE`, they should technically be mandatory for v2+.

[the remark this answers is not in the document]
Comment (mattklein123) [reply include/envoy/server/filter_config.h]: +1

**Comment by mattklein123** [comment]
Let's do this!!! So excited. LMK when it's ready for proper review.

/wait

**Comment by alyssawilk** [comment]
+1, release is now cut, so feel free to do any final clean up and de-draft this PR!

**Comment by derekargueta** [comment]
@alyssawilk good to go from me - was just making sure all builds passed

**Comment by mattklein123** [comment]
:tada: :tada: :tada: 

You are a hero!

**Pull request merged**